mos_tests/conftest.py

```diff
@@ -269,14 +269,40 @@ def is_l3_ha(env):
 
 @pytest.fixture(autouse=True)
 def env_requirements(request, env):
+    reserved = {'or', 'and', 'not', '(', ')'}
     marker = request.node.get_marker('check_env_')
-    if marker:
-        for func_name in marker.args:
-            func = globals().get(func_name)
-            if func is not None and not func(env):
-                doc = func.__doc__ or 'Env {}'.format(
-                    func_name.replace('_', ' '))
-                pytest.skip('Requires: {}'.format(doc))
+    if not marker:
+        return
+    marker_str = marker.args[0]
+    if len(marker.args) > 1:
```

Comment:
Sorry, i'm not noticed next 2 lines.
But you don't need to retrive first arg, check line of marker args, `marker_str = ' and '.join(marker.args)` give to you same result